src/runtime/runtime_nintendoswitch_heap.go

```diff
@@ -24,7 +19,12 @@ func preinit() {
 	setHeapSize(&heapStart, heapSize)
 
 	if heapStart == 0 {
-		panic("failed to allocate heap")
+		// Panic should work, but if we call panic here, it triggers a compiler bug that breaks
+		// all print calls in the whole application. So far we couldn't figure out why, that's
+		// why we're using a byte array and nxOutputString directly
+		nxOutputString(&failedToAllocateHeapString[0], uint64(len(failedToAllocateHeapString)))
+		exit(1)
+		//panic("failed to allocate heap")
```

Comment:
This appears unrelated, when I revert this change it still works. However, it would be a good idea to use a `runtimePanic` instead which doesn't allocate (assuming that `putchar` doesn't allocate):

```suggestion
		runtimePanic("failed to allocate heap")
```

Reply:
Great! Any major differences between runtimePanic vs panic?

Reply:
Yes, `runtimePanic` is somewhat lighter weight and should generally not do heap allocations (which is important if you're trying to panic when memory is full, for example). The `panic` function might call an interface method that allocates, so the compiler can't prove it won't allocate.